packages/x-date-pickers/src/internals/components/PickerFieldUI.tsx

```diff
@@ -369,7 +369,7 @@ export interface PickerFieldUISlotsFromContext extends PickerFieldUISlots {
 
 export interface PickerFieldUISlotProps {
   textField?: SlotComponentPropsFromProps<
-    PickersTextFieldProps | TextFieldProps,
+    Omit<TextFieldProps, 'onKeyDown'> | PickersTextFieldProps,
```

Comment:
This condition seems slightly buggy, it looks like only the last entry gets picked up... 🙈 🤷 
Moved types around and omitted the plain `onKeyDown` handler from the Mui component types.